### PR TITLE
TEST-#4879: Use pandas `ensure_clean()` in place of `io_tests_data`

### LINKED ISSUE #4879: TST: temp files left behind after pytest modin/pandas/test/
I see 200kx99.csv and modin/pandas/test/io_tests_data/ left behind.

could reuse pandas' tm.ensure_clean for this

### PR DESCRIPTION
**Pull request by pyrito**
## What do these changes do?

This PR does some test cleanup to ensure that `io_tests_data` gets cleaned up after pytests finish. There was also another lingering file from another test that was changed to use the `get_unique_filename` method.

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4879 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

**Review by mvashishtha — DISMISSED** [review]

**Review by mvashishtha — APPROVED** [review]

**Review by jbrockmendel — APPROVED** [review]
LGTM

File: modin/conftest.py [hunk 1]
```diff
@@ -545,6 +541,10 @@ def get_generated_doc_urls():
 
 
 def pytest_sessionstart(session):
+    # create test data dir if it is not exists yet
+    if not os.path.exists(IO_OPS_DATA_DIR):
+        os.mkdir(IO_OPS_DATA_DIR)
```
Comment (vnlitvinov) [inline modin/conftest.py:546]:
```suggestion
    os.makedirs(IO_OPS_DATA_DIR, exist_ok=True)
```

File: modin/conftest.py [hunk 2]
```diff
@@ -556,6 +556,10 @@ def pytest_sessionstart(session):
 
 
 def pytest_sessionfinish(session, exitstatus):
+    # Cleanup io_tests_data
+    if os.path.exists(IO_OPS_DATA_DIR):
+        shutil.rmtree(IO_OPS_DATA_DIR)
```
Comment (vnlitvinov) [inline modin/conftest.py:561]:
```suggestion
    shutil.rmtree(IO_OPS_DATA_DIR, ignore_errors=True)
```

**Review by vnlitvinov — COMMENTED** [review]
The only foreseeable issue is if someone is running two sessions at once in the same directory...

Why did it hurt to keep that directory anyway?

**Comment by pyrito** [comment]
@vnlitvinov you are right, it doesn't hurt to keep the dir but it looks like with pytest-xdist we can have race conditions which can lead to one worker deleting the director and leave the other trying to write to it. You can see here: https://github.com/pytest-dev/pytest/issues/503. It's a bummer that we can't have a global setup and teardown for pytest-xdist.

I'm now thinking of changing all our use cases of `get_unique_filename` to use a temporary directory within a context manager instead. Thoughts?

**Comment by jbrockmendel** [comment]
> I'm now thinking of changing all our use cases of get_unique_filename to use a temporary directory within a context manager instead. Thoughts?

+1.  re-use ensure_clean

File: modin/experimental/pandas/test/test_io_exp.py [hunk 3]
```diff
@@ -259,47 +259,45 @@ def test_distributed_pickling(filename, compression):
...
+            df = pandas.DataFrame(data, columns=[f"col{x}" for x in range(ncols)])
+            df.to_json(file_name, lines=True, orient="records")
+
+        _generate_json(filename, 64, 8)
+
+        # Custom parser allows us to add some specifics to reading files,
+        # which is not available through the ready-made API.
+        # For example, the parser allows us to reduce the amount of RAM
+        # required for reading by selecting a subset of columns.
+        def _custom_parser(io_input, **kwargs):
```
Comment (jbrockmendel) [inline modin/experimental/pandas/test/test_io_exp.py:275]: can this go outside the context?

File: modin/pandas/test/test_io.py [hunk 4]
```diff
@@ -1961,41 +1964,42 @@ def test_read_hdf(self, make_hdf_file, format):
...
-                store.append("data/df2", pd.DataFrame(np.random.randn(4, 4)))
-
-            modin_df = pd.read_hdf(hdf_file, key="data/df1", mode="r")
-            pandas_df = pandas.read_hdf(hdf_file, key="data/df1", mode="r")
-            df_equals(modin_df, pandas_df)
-        finally:
-            if hdf_file:
-                os.unlink(hdf_file)
-            teardown_test_files([unique_filename_modin, unique_filename_pandas])
+        with ensure_clean_dir() as dir:
```
Comment (jbrockmendel) [inline modin/pandas/test/test_io.py:1967]: `dir` -> `dirname`?

File: modin/pandas/test/test_io.py [hunk 5]
```diff
@@ -1904,21 +1906,22 @@ def test_ExcelFile(self, make_excel_file):
...
-            pandas_writer.save()
+                modin_writer.save()
+                pandas_writer.save()
 
-            assert assert_files_eq(unique_filename_modin, unique_filename_pandas)
-        finally:
-            teardown_test_files([unique_filename_modin, unique_filename_pandas])
+                assert assert_files_eq(unique_filename_modin, unique_filename_pandas)
+            finally:
+                teardown_test_files([unique_filename_modin, unique_filename_pandas])
```
Comment (jbrockmendel) [inline modin/pandas/test/test_io.py:1924]: is this still necessary?

File: modin/pandas/test/test_io.py [hunk 6]
```diff
@@ -1587,30 +1588,31 @@ def test_read_parquet_without_metadata(self, engine):
...
-            # read into pyarrow table and write it to a parquet file
-            t = csv.read_csv(csv_fname)
-            parquet.write_table(t, parquet_fname)
-
-            eval_io(
-                "read_parquet",
-                # read_parquet kwargs
-                path=parquet_fname,
-                engine=engine,
+        with ensure_clean_dir() as dir:
```
Comment (jbrockmendel) [inline modin/pandas/test/test_io.py:1591]: dir->dirname

File: modin/pandas/test/test_io.py [hunk 7]
```diff
@@ -1526,7 +1528,8 @@ def test_read_parquet_pandas_index_partitioned(self, engine):
                 "C": ["c"] * 2000,
             }
         )
-        try:
+        with ensure_clean_dir() as dir:
```
Comment (jbrockmendel) [inline modin/pandas/test/test_io.py:1531]: dir->dirname

File: modin/pandas/test/test_io.py (hunk 5 above)
Comment (pyrito) [reply modin/pandas/test/test_io.py:1932]: Nope

File: modin/experimental/pandas/test/test_io_exp.py [hunk 8]
```diff
@@ -37,51 +38,53 @@
...
-            conn,
-            partition_column="col1",
-            lower_bound=0,
-            upper_bound=6,
-            max_sessions=2,
-        )
-
-        df_equals(modin_df_from_query, pandas_df)
-        df_equals(modin_df_from_table, pandas_df)
+            df_equals(modin_df_from_query, pandas_df)
```
Comment (jbrockmendel) [inline modin/experimental/pandas/test/test_io_exp.py:65]: can these go outside the context?

File: modin/experimental/pandas/test/test_io_exp.py [hunk 9]
```diff
@@ -37,51 +38,53 @@
...
-        modin_df_from_table = pd.read_sql(table, conn)
+        pandas_df = pandas.read_sql(query, conn)
+        with pytest.warns(UserWarning):
+            modin_df_from_query = pd.read_sql(query, conn)
+        with pytest.warns(UserWarning):
+            modin_df_from_table = pd.read_sql(table, conn)
 
-    df_equals(modin_df_from_query, pandas_df)
-    df_equals(modin_df_from_table, pandas_df)
+        df_equals(modin_df_from_query, pandas_df)
```
Comment (jbrockmendel) [inline modin/experimental/pandas/test/test_io_exp.py:86]: ditto can these go outside the context

File: modin/pandas/test/test_io.py [hunk 10]
```diff
@@ -1381,12 +1380,13 @@ def test_read_parquet_indexing_by_column(self, make_parquet_file):
...
+        with ensure_clean_dir() as dirname:
+            unique_filename = get_unique_filename(extension="parquet", data_dir=dirname)
+            make_parquet_file(filename=unique_filename, nrows=nrows)
 
-        parquet_df = pd.read_parquet(unique_filename)
-        for col in parquet_df.columns:
-            parquet_df[col]
+            parquet_df = pd.read_parquet(unique_filename)
+            for col in parquet_df.columns:
+                parquet_df[col]
```
Comment (mvashishtha) [inline modin/pandas/test/test_io.py:1389]:
```suggestion
        for col in parquet_df.columns:
            parquet_df[col]
```

**Review by mvashishtha — APPROVED** [review]
Thanks @pyrito ! LGTM with one minor comment.

**Review by mvashishtha — APPROVED** [review]

**Review by vnlitvinov — COMMENTED** [review]
Can we drop `get_unique_filename()` completely?

**Comment by pyrito** [comment]
@vnlitvinov there are still some cases where it's useful to have around. We can make that a separate task to clean up, I don't think it needs to get rolled into this PR since the diffs are already quite big.

**Review by vnlitvinov — APPROVED** [review]
LGTM!

**Review by jbrockmendel — APPROVED** [review]
LGTM

**Pull request merged**